server/src-lib/Hasura/GraphQL/Transport/WebSocket.hs

```diff
@@ -517,24 +529,26 @@ onMessage
   -> AuthMode
   -> WSServerEnv
   -> WSConn -> BL.ByteString -> m ()
-onMessage env authMode serverEnv wsConn msgRaw = Tracing.runTraceT "websocket" do
-  case J.eitherDecode msgRaw of
-    Left e    -> do
-      let err = ConnErrMsg $ "parsing ClientMessage failed: " <> T.pack e
-      logWSEvent logger wsConn $ EConnErr err
-      sendMsg wsConn $ SMConnErr err
-
-    Right msg -> case msg of
-      CMConnInit params -> onConnInit (_wseLogger serverEnv)
-                           (_wseHManager serverEnv)
-                           wsConn authMode params
-      CMStart startMsg  -> onStart env serverEnv wsConn startMsg
-      CMStop stopMsg    -> liftIO $ onStop serverEnv wsConn stopMsg
-      -- The idea is cleanup will be handled by 'onClose', but...
-      -- NOTE: we need to close the websocket connection when we receive the
-      -- CMConnTerm message and calling WS.closeConn will definitely throw an
-      -- exception, but I'm not sure if 'closeConn' is the correct thing here....
-      CMConnTerm        -> liftIO $ WS.closeConn wsConn "GQL_CONNECTION_TERMINATE received"
+onMessage env authMode serverEnv wsConn msgRaw = do
+  L.unLogger logger $ WSMessageLog (BL.length msgRaw)
```

Comment:
This will materialize the message in memory. I'm not sure if we care about this as a possible source of space leaks, since I know we do this sort of thing in plenty of other places already. I don't think this needs to block anything here. cc @jberryman what do you think?

Reply:
I think it's probably okay too, but also I wonder: lazy ByteStrings don't have to imply lazy IO, which is happening here? I assume that's what you meant about `BL.length`: that it would be forcing chunks into memory?

Reply:
Yes, I suppose that's right, we're not doing lazy IO necessarily. I think this is fine, thanks!

Reply:
Thanks for all your comments :)